lib/cosmos/tools/cmd_tlm_server/cmd_tlm_server_gui.rb

```diff
@@ -320,6 +320,57 @@ def graceful_kill
       # Just to avoid warning
     end
 
+    def self.graceful_kill
+      # Just to avoid warning
+    end
+
+    def self.no_gui_handle_string_output
+      if @string_output.string[-1..-1] == "\n"
+        lines_to_write = ''
+        string = @string_output.string.clone
+        @string_output.string = @string_output.string[string.length..-1]
+        string.each_line {|out_line| lines_to_write << out_line }
+        @message_log.write(lines_to_write)
+        STDOUT.print lines_to_write if STDIN.isatty # Have a console
+      end
+    end
+
+    def self.no_gui_stop_callback
+      no_gui_handle_string_output()
+      @output_sleeper.cancel
+      Cosmos.kill_thread(self, @output_thread)
+      no_gui_handle_string_output()
+    end
+
+    def self.post_options_parsed_hook(options)
+      if options.no_gui
+        begin
+          @output_sleeper = Sleeper.new
+          @string_output = StringIO.new("", "r+")
+          $stdout = @string_output
+          Logger.level = Logger::INFO
+          cts = CmdTlmServer.new(options.config_file, options.production)
+          @message_log = CmdTlmServer.message_log
+          @output_thread = Thread.new do
+            while true
+              no_gui_handle_string_output()
+              break if @output_sleeper.sleep(1)
+            end
+          end
+          cts.stop_callback = method(:no_gui_stop_callback)
+          sleep # Sleep until waked by signal
+        ensure
+          if defined? cts and cts
```

Comment:
Restore the check of defined? but left in also making sure it is set.   The defined? check is needed in case something goes wrong before it is defined.